src/main/kotlin/com/example/taskManagerController/controller/TaskController.kt

```
@@ -0,0 +1,52 @@
+    fun getTaskById(@PathVariable id: UUID): ResponseEntity<Task> {
+        return taskService.getTaskById(id)
+            ?.let { task -> ResponseEntity.ok(task) }
+            ?: ResponseEntity.notFound().build()
+    }
+
+    @PostMapping
+    fun createTask(@RequestBody task: Task): ResponseEntity<Task?> {
+        return try {
+            val createdTask = taskService.addTask(task)
+            ResponseEntity.status(HttpStatus.CREATED).body(createdTask)
+        } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)